rfcs/0032-blockchain-in-hard-fork

```diff
@@ -0,0 +1,153 @@
+## Summary
+[summary]: #summary
+
+We present alternatives for handling the blockchain in a hard fork. The focus is
+on hard forks in response to a network failure, though the mechanisms may be
+applicable in more general scenarios.
+
+This RFC does not consider how the transaction SNARK and scan state are handled
+in hard forks. There will be another RFC for those, and an RFC to describe
+the entire hard fork plan.
+
+## Motivation
+[motivation]: #motivation
+
+For the launch of the Coda main net, we wish to have a hard fork plan and implementation
+in place, in case there's a network failure. One of the parts of such a plan
+is how to update the blockchain.
+
+## Summary description of alternatives
+[summary-descriptions]: #summary-descriptions
+
+After the fork, do one of these:
+
+1. New genesis timestamp and genesis ledger; assert a new starting epoch and slot
+
+2. Retain the original genesis timestamp, with no time offset, with rules to
+    maintain chain strength
+
+3. Retain the original genesis timestamp, use a time offset for new blocks
+
+## Detailed description of alternatives
+[detailed-descriptions]: #detailed-descriptions
+
+Alternative 1
+=============
+
+- choose a root and ledger at the pause point
+
+- the initial epoch, slot are the special case of 0,0
+   - otherwise, increment the slot from the pause point
+   - provide as a compile-time parameter
+
+- if we have a proof of the protocol state, provide that
+
+- subsequent blocks may use different proof, verification keys
+
+- force a software upgrade by bumping the protocol major version
+
+Notes
+-----
+
+- nodes will bootstrap from the new genesis state
+
+- time-locked accounts and transactions with expiration should continue
+   to work, because they rely on global slot numbers, and the
+   slot is incremented from the pause point
+
+Issues
+------
+
+- what is the source of randomness (epoch seed) for the post-fork chain?
+
+- is it acceptable to erase history (but we already have a succinct blockchain)
+
+- specifically, is there unfairness to the chains discarded past the pause point
+
+- what is allowed to change after the fork, with what effect:
+   - protocol state structure
+   - ledger structure/depth,
+   - account structure
+   - consensus constants
+
+- do we have to maintain pre-, post-fork code/data structures to verify the new genesis state,
+   ledger, and new blocks
+
+- if there's no proof of the ledger, how to convince the world that the new genesis ledger and
```

Comment:
I think we would have to have an archive functionality that saved the protocol state at the breakpoint, and write a verification procedure that checked consistency with the state after the break point